### ISSUE TITLE
Expand support for configuration.

### ISSUE THREAD

**Issue by jxnu-liguobin** [kobylynskyi/graphql-java-codegen#585]
It seems that JSON doesn't support configure merging. As a client of automation, I need this function.

I know that in Scala, **[typesafe conf](https://github.com/lightbend/config#java-yep-wrappers-for-the-java-library)(value as object, so there is no need to escape, and almost all JVM languages are supported)** can support the merging of multiple files. In this way, we can use the predefined configuration and custom configuration, such as the serialization of the client. We can achieve this by adding custom configuration instead of directly configuring when writing the client.

example:
client has a config:
```json
{
  "generateClient": true,
  "generateApis": true,
  "generateBuilder": true,
  "generateImmutableModels": true,
  "generateToString": true,
  "generateEqualsAndHashCode": true,
  "apiPackageName": "io.github.graphql.j.resolver",
  "modelPackageName": "io.github.graphql.j.model",
  "generateModelOpenClasses": true,
  "modelNameSuffix": "TO",
  "apiInterfaceStrategy": "DO_NOT_GENERATE",
  "apiRootInterfaceStrategy": "SINGLE_INTERFACE",
  "generateModelsForRootTypes": "true",
  "apiNamePrefix": "GitHub",
  "addGeneratedAnnotation": false,
  "generatedLanguage": "JAVA"
}
```

someone use client:
add custom config to classpath, then codegen plugin read and merge it.
```json
{
  ......
}
```

Of course, I'm not sure if the plug-in can read files when it performs tasks. But if accepted, I will consider and test for it.

**Comment by kobylynskyi** [comment]
@jxnu-liguobin if I understand your request correctly, we already support configuration from the .json file + configuration in the build file (pom.xml / build.gradle / etc)
So you can already specify all codegen configs in json.
Please explain a bit more if I misunderstood you. Thanks.

**Comment by jxnu-liguobin** [comment]
> @jxnu-liguobin if I understand your request correctly, we already support configuration from the .json file + configuration in the build file (pom.xml / build.gradle / etc)
> So you can already specify all codegen configs in json.
> Please explain a bit more if I misunderstood you. Thanks.

I want to use JSON as the default and another JSON file to config

**Comment by kobylynskyi** [comment]
@jxnu-liguobin, merged to develop. Will be released as part of 5.0.0.
